### PR TITLE
[Android] Upgrade setup-java action and enable gradle cache

### PR DESCRIPTION
**Pull request by Khufu-I**

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -15,10 +15,11 @@ jobs:
         working-directory: ./android
     steps:
       - uses: actions/checkout@v3
-      # Upgrade Github's  default Java 1.8
-      - uses: actions/setup-java@v1
+      - uses: actions/setup-java@v3
         with:
-          java-version: '17'
+          distribution: zulu
```
Comment (Vithop) [inline .github/workflows/android.yml:20]: what's zulu?

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -15,10 +15,11 @@ jobs:
...
+          java-version: 17
```
Comment (Vithop) [inline .github/workflows/android.yml:21]: why not v19?

File: .github/workflows/android.yml (hunk 1 above)
Comment (Khufu-I) [reply .github/workflows/android.yml:20]: It’s the [open idk distribution](https://github.com/actions/setup-java#supported-distributions)

File: .github/workflows/android.yml (hunk 2 above)
Comment (Khufu-I) [reply .github/workflows/android.yml:21]: The [docs say 17 is the highest](https://github.com/actions/setup-java#supported-version-syntax) but I guess I should try 19

**Review by Vithop — APPROVED** [review]

**Pull request merged**